src/providers/ApiProvider.tsx

```diff
@@ -0,0 +1,18 @@
+import React, { ReactNode } from 'react';
+import { useDispatch } from 'react-redux';
+import { SOCKET_IO, ICurrentUserJoinRoomData, currentUserJoinRoom } from '../store';
+import { useSocket } from '../hooks';
+
+export interface IApiProviderProps {
+  children: ReactNode;
+}
+
+export function ApiProvider({ children }: IApiProviderProps) {
+  const dispatch = useDispatch();
+  
+  useSocket<ICurrentUserJoinRoomData>(SOCKET_IO.ON_CURRENT_USER_JOIN_ROOM, (data) => {
```

Comment:
This file will listen for events emitted by the server and essentially convert these to redux actions. These actions will be picked up by the sagas which will handle updating state.